### PR TITLE
IronResizableBehavior added

### PR DESCRIPTION
**Pull request by MartinMoizard**
I added IronResizableBehavior in order to notify main that it's content has been resized. It can happen for example when this.narrow changes.

### CODE REVIEW

**Comment by MartinMoizard** [comment]
Is it possible to have some details on the reasons why tests are failing? Thank you.

**Comment by blasten** [comment]
What's the use case here? It seems like you are using a custom transition.

**Comment by MartinMoizard** [comment]
My paper-drawer-panel wraps a component implementing IronResizableBehavior. So when I toggle forceNarrow on my paper-drawer-panel, the size of the `main` div changes (through _computeMainStyle): the css left value changes, then an animation is triggered and I want to be notified that the size changed when the animation finishes.

**Comment by blasten** [comment]
I see, I think this is reasonable. cc @frankiefu any comment?

**Comment by MartinMoizard** [comment]
Hello @blasten, @frankiefu,

Any update on this PR?

Thanks

File: paper-drawer-panel.html [hunk 1]
```diff
@@ -393,6 +396,22 @@
...
 
+      attached: function() {
+        this.listen(this.$.main, 'transitionend', '_transitionEndCallback');
+        this.listen(this.$.main, 'webkitTransitionEnd', '_transitionEndCallback');
+      },
+
+      detached: function() {
+        this.unlisten(this.$.main, 'transitionend', '_transitionEndCallback');
+        this.unlisten(this.$.main, 'webkitTransitionEnd', '_transitionEndCallback');
+      },
```
Comment (frankiefu) [inline paper-drawer-panel.html]: I think you can remove the above `listen` and `unlisten` and just do `on-<event>` annotation syntax.  e.g.

```
<div id="main" on-transitionend="_transitionEndCallback">
```

Comment (blasten) [inline paper-drawer-panel.html]: +1

File: paper-drawer-panel.html [hunk 2]
```diff
@@ -393,6 +396,22 @@
...
+
+      _transitionEndCallback: function (e) {
```
Comment (frankiefu) [inline paper-drawer-panel.html]: Can you change the method name to something like `_onMainTransitionEnd` so we know is for the transitionEnd on the #main?

**Comment by MartinMoizard** [comment]
@frankiefu thanks for your feedbacks. I pushed a new version.

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**